influence_benchmark/stats/utils_pandas.py

```diff
@@ -0,0 +1,102 @@
+"""
+This file contains functions which represent the collected
+data as pandas dataframes at different levels of granularity 
+(turns, trajectories, initial_states).
+"""
+
+from collections import defaultdict
+from pathlib import Path
+from typing import Dict, List, Tuple, Union
+
+import pandas as pd
+
+from influence_benchmark.root import PROJECT_DATA
+
+
+def calculate_expectation(score_distribution: Dict[str, float]) -> float:
+    """Calculate the expected preference rating or expected influence rating from a single set of preferences."""
+    return sum(float(score) * probability for score, probability in score_distribution.items())
+
+
+def load_turns_df_from_traj_path(trajectory_path: Path) -> pd.DataFrame:
+    # Read all trajectories from files
+    turns_df = pd.concat([pd.read_json(file, lines=True) for file in trajectory_path.glob("[0-9]*.jsonl")])
+
+    # Calculate expected preference
+    turns_df["timestep_reward"] = turns_df["preferences"].apply(calculate_expectation)
+    if "influence_scores" in turns_df.columns:
+        turns_df["timestep_influence_level"] = turns_df["influence_scores"].apply(calculate_expectation)
+    else:  # for backwards compatibility
+        turns_df["timestep_influence_level"] = 0
+    return turns_df
+
+
+def group_turns_df_to_traj_df_final(turns_df):
+    # Get the final turn for each trajectory
+    traj_final_df = (
+        turns_df.groupby(["env_name", "initial_state_id", "trajectory_id"])
+        .apply(lambda x: x.loc[x["turn"].idxmax()])
+        .reset_index(drop=True)
+    )
+
+    # Select the reward and influence level from the final turn
+    traj_final_df = traj_final_df[
+        ["env_name", "initial_state_id", "trajectory_id", "timestep_reward", "timestep_influence_level"]
+    ].rename(columns={"timestep_reward": "traj_final_rew", "timestep_influence_level": "traj_final_infl"})
+
+    return traj_final_df
+
+
+def group_turns_df_to_traj_df(turns_df):
+    # Average over turns, will include num_envs * num_initial_states * num_trajs_per_initial_state rows
+    traj_df = (
+        turns_df.groupby(["env_name", "initial_state_id", "trajectory_id"])[
+            ["timestep_reward", "timestep_influence_level"]
+        ]
+        .mean()
+        .reset_index()
+        .rename(columns={"timestep_reward": "traj_mean_rew", "timestep_influence_level": "traj_mean_infl"})
+    )
+    return traj_df
```

Comment:
I feel like the names for these functions are a bit uninformative. Seems like the point here is to add reward/influence info, but that isn't part of the name at all

Reply:
for doing this kinds of renaming tasks, you'll probably find fn+F2 to be really helpful after clicking on the method name